launcher/ui/pages/instance/StoragePage.cpp

```diff
@@ -0,0 +1,158 @@
+    m_series->setLabelsVisible();
+    m_series->setLabelsPosition(QtCharts::QPieSlice::LabelInsideHorizontal);
+    m_chart_view = new QtCharts::QChartView();
+    m_chart = new QtCharts::QChart();
+    m_chart->addSeries(m_series);
+    m_chart->setBackgroundVisible(false);
+    m_chart->setMargins(QMargins(0, 0, 0, 0));
+    m_chart->legend()->setAlignment(Qt::AlignLeft);
+    m_chart_view->setRenderHint(QPainter::Antialiasing);
+
+    ui->verticalLayout->addWidget(m_chart_view);
+    ui->retranslateUi(this);
+
+    updateCalculations();
+
+    connect(ui->button_goto_resouce_packs, &QPushButton::clicked, this, [&] { m_container->selectPage("resourcepacks"); });
+    connect(ui->button_goto_mods, &QPushButton::clicked, this, [&] { m_container->selectPage("mods"); });
+    connect(ui->button_goto_worlds, &QPushButton::clicked, this, [&] { m_container->selectPage("worlds"); });
+    connect(ui->button_goto_screenshots, &QPushButton::clicked, this, [&] { m_container->selectPage("screenshots"); });
+    connect(ui->button_goto_other_logs, &QPushButton::clicked, this, [&] { m_container->selectPage("logs"); });
+
+    connect(ui->button_clear_screenshots, &QPushButton::clicked, this, &StoragePage::handleClearScreenshotsButton);
+    connect(ui->button_clear_logs, &QPushButton::clicked, this, &StoragePage::handleClearLogsButton);
+    connect(ui->button_clear_all, &QPushButton::clicked, this, &StoragePage::handleClearAllButton);
+}
+
+StoragePage::~StoragePage()
+{
+    delete ui;
+}
+
+bool StoragePage::apply()
+{
+    return true;
+}
+
+void StoragePage::retranslate()
+{
+    ui->retranslateUi(this);
+}
+
+void StoragePage::handleClearScreenshotsButton()
+{
+    if (m_confirmation_box->exec() != QMessageBox::Yes)
+        return;
+
+    auto path = m_inst->gameRoot() + "/screenshots";
+    QDir(path).removeRecursively();
+    updateCalculations();
+}
+
+void StoragePage::handleClearLogsButton()
+{
+    if (m_confirmation_box->exec() != QMessageBox::Yes)
+        return;
+
+    auto path = m_inst->gameRoot() + "/logs";
+    QDir(path).removeRecursively();
+    updateCalculations();
+}
+
+void StoragePage::handleClearAllButton()
+{
+    if (m_confirmation_box->exec() != QMessageBox::Yes)
+        return;
+
+    handleClearScreenshotsButton();
+    handleClearLogsButton();
+}
+
+void StoragePage::updateCalculations()
+{
+    auto size_resource_packs =
+        getDirectorySize((m_inst->gameRoot() + "/texturepacks")) + getDirectorySize((m_inst->gameRoot() + "/resourcepacks"));
+    auto size_mods = getDirectorySize(m_inst->modsRoot());
+    auto size_saves = getDirectorySize((m_inst->gameRoot() + "/saves"));
+    auto size_screenshots = getDirectorySize((m_inst->gameRoot() + "/screenshots"));
+    auto size_logs = getDirectorySize((m_inst->gameRoot() + "/logs"));
+
+    auto storage_info = QStorageInfo(QDir(m_inst->gameRoot()));
+    auto size_remaining = storage_info.bytesAvailable();
+    auto size_used = storage_info.bytesTotal() - size_remaining;
```

Comment:
Surely this gives you the size used on the ENTIRE storage device that the instance is on, rather than the size used just by the instance. Is that intentional?